mrmustard/utils/graphics.py

```diff
@@ -64,45 +65,70 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         return self.bar.__exit__(exc_type, exc_val, exc_tb)
 
 
-def mikkel_plot(rho: np.ndarray, xbounds: Tuple[int] = (-6, 6), ybounds: Tuple[int] = (-6, 6)):
+def mikkel_plot(
+    rho: np.ndarray,
+    xbounds: Tuple[int] = (-6, 6),
+    ybounds: Tuple[int] = (-6, 6),
+    xticks=(-5, 0, 5),
+    xtick_labels=None,
+    yticks=(-5, 0, 5),
+    ytick_labels=None,
+    grid=False,
+    resolution: int = 200,
+):  # pylint: disable=too-many-statements
     """Plots the Wigner function of a state given its density matrix.
 
     Args:
         rho (np.ndarray): density matrix of the state
         xbounds (Tuple[int]): range of the x axis
         ybounds (Tuple[int]): range of the y axis
+        xticks (Tuple[int]): ticks of the x axis
+        xtick_labels (Optional[Tuple[str]]): labels of the x axis; if None uses default formatter
+        yticks (Tuple[int]): ticks of the y axis
+        ytick_labels (Optional[Tuple[str]]): labels of the y axis; if None uses default formatter
+        grid (bool): whether to display the grid
+        resolution (int): resolution of the plot
+
+    Returns:
+        tuple: figure and axes
     """
 
-    xvec = np.linspace(*xbounds, 200)
-    pvec = np.linspace(*ybounds, 200)
+    q, ProbX = quadrature_distribution(rho)
+    p, ProbP = quadrature_distribution(rho, np.pi / 2)
+
+    xvec = np.linspace(*xbounds, resolution)
+    pvec = np.linspace(*ybounds, resolution)
     W, X, P = wigner_discretized(rho, xvec, pvec, settings.HBAR)
-    ProbX = np.sum(W, axis=1)
-    ProbP = np.sum(W, axis=0)
 
     ### PLOTTING ###
 
-    _, ax = plt.subplots(
+    fig, ax = plt.subplots(
         2, 2, figsize=(6, 6), gridspec_kw={"width_ratios": [2, 1], "height_ratios": [1, 2]}
     )
-    ticks = [-5, 0, 5]
-    grid = False
     plt.subplots_adjust(wspace=0.05, hspace=0.05)
 
+    if xtick_labels is None:
+        xtick_labels = xticks
+    if ytick_labels is None:
+        ytick_labels = yticks
+
     # Wigner function
-    ax[1][0].contourf(X, P, W, 60, cmap=cm.RdBu, vmin=-abs(W).max(), vmax=abs(W).max())
+    ax[1][0].contourf(X, P, W, 120, cmap=cm.RdBu, vmin=-abs(W).max(), vmax=abs(W).max())
```

Comment:
```suggestion
    ax[1][0].contourf(X, P, W, 120, cmap=cm.magma, vmin=-abs(W).max(), vmax=abs(W).max())
```
I think we should prefer a perceptually uniform sequential colormap https://matplotlib.org/stable/tutorials/colors/colormaps.html.